### PR TITLE
[opentelemetry-operator] bugfixing, bump versions, update crds to beta

### PR DESCRIPTION
**Pull request by timojohlo**
## Pull Request Details

- Updating CR to v1beta1 (following the CRD update) 
- Add a worthy description to the customizing fields in the plugin definition
- Disable `logs/k8sevents` for the moment, as this is currently incomplete (missing processing pipeline)
- Updating tar chart file
- Bumping chart version to next stable version (0.2.0)

### CODE REVIEW

File: opentelemetry-operator/chart/Chart.yaml [hunk 1]
```diff
@@ -2,9 +2,9 @@
 # SPDX-License-Identifier: Apache-2.0
 
 apiVersion: v2
-appVersion: v0.104.0
+appVersion: v0.103.0
```
Comment (viennaa) [inline opentelemetry-operator/chart/Chart.yaml:5]: is this switch back intended?

Comment (timojohlo) [reply opentelemetry-operator/chart/Chart.yaml]: Thanks for pointing out, fixed!

File: opentelemetry-operator/chart/templates/metrics-collector.yaml [hunk 2]
```diff
@@ -3,21 +3,26 @@ SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse c
...
-apiVersion: opentelemetry.io/v1alpha1
+apiVersion: opentelemetry.io/v1beta1
 kind: OpenTelemetryCollector
 metadata:
   name: metrics 
 spec:
   mode: deployment
   image: {{ index .Values "opentelemetry-operator" "manager" "collectorImage" "repository" }}:{{ index .Values "opentelemetry-operator" "manager" "collectorImage" "tag" }}
-  config: |
+  config:
```
[the remark this answers is not in the document]
Comment (timojohlo) [reply opentelemetry-operator/chart/templates/metrics-collector.yaml:13]: No, it actually throws an error with pipe (`|`) as prefix: `cannot unmarshal string into Go struct field OpenTelemetryCollectorSpec.spec.config of type v1beta1.Config`

[the remark this answers is not in the document]
Comment (viennaa) [reply opentelemetry-operator/chart/templates/metrics-collector.yaml:13]: great. it has changed with v1beta to not have this "object dump" there anymore.

**Review by viennaa — APPROVED** [review]
Alright, good to go. Thanks.

**Pull request merged**